src/lib/registration/controllers/addons_selection.rb

```diff
@@ -0,0 +1,259 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "registration/addon"
+require "registration/addon_sorter"
+
+module Registration
+  module Controllers
+    # Controller to manage actions and state of addon selection
+    #
+    # Needed by {Widgets::MasterDetailSelector}
+    class AddonsSelection
+      include Yast::I18n
+
+      class << self
+        # @return [Boolean] If not released (e.g., alpha or beta versions) addons should be filtered
+        attr_accessor :filtering_unreleased
+      end
+
+      # A checkbox item representation, see {Widgets::CheckboxItem}
+      Item = Struct.new(:id, :label, :status, :description)
```

Comment:
This is a bit confusing as Item is also in Yast::UIShortcuts. I would choose a different constant name.